joomla-framework/string
# Test consolidation

### Reorganisation and completion of tests

This package still uses `phputf8`, which has long been abandoned.
In preparation of an exchange (see #29), this PR increases the test coverage from ~75% to >96%.
A few small issues have necessarily been solved on the way.

### Summary of Changes

#### Tests
- Add test cases
- Improve testdox annotations
- Add type hints

#### StringHelper
- Reduce complexity to increase path/branch coverage

#### Inflector
- Remove internal use of deprecated code

#### phputf8
- Fix bug in str_ireplace() - count variable was not populated

### Testing Instructions

- Code Review. Make sure that signatures in StringHelper and Inflector did not change in an incompatible way. Changes to signatures should only affect default values.

### Documentation Changes Required

None.
